### DISCUSSION TITLE
BUILD FAILED (macOS 14.5 on arm64 using ruby-build 20240517) OpenSSL is installed still not able to find

### DISCUSSION THREAD

**Discussion by DemonJAZ** [rbenv/ruby-build#2393] [Build failures] [answered]
i am trying to install ruby 3.3.1 using asdf , all the dependencies like openssl are installed but i am still not able to find any fix
![image](https://github.com/rbenv/ruby-build/assets/14571179/17184077-31ea-45c5-834c-117f8a02e545)

in the log file i am getting:
ln -sf ../../rbconfig.rb .ext/arm64-darwin23/rbconfig.rb
linking ruby
*** Following extensions are not compiled:
openssl:
	Could not be configured. It will not be installed.
	/private/var/folders/2z/pc9j54vs7jdb271vz91fnk400000gn/T/ruby-build.20240527125744.25384.etdrLh/ruby-3.3.1/ext/openssl/extconf.rb:122: OpenSSL library could not be found. You might want to use --with-openssl-dir=<dir> option to specify the prefix where OpenSSL is installed.
	Check ext/openssl/mkmf.log for more details.
psych:
	Could not be configured. It will not be installed.
	Check ext/psych/mkmf.log for more details.
*** Fix the problems, then remove these directories and try again if you want.

Attaching complete log file for details
[Installlog.log](https://github.com/rbenv/ruby-build/files/15452541/Installlog.log)

Please Help

**Comment by DemonJAZ** [comment, marked as answer]
i found the solution , seems like since i migrated my mac from intel to apple silicon , it was using intel arch installed homebrew

**Reply by riturajratan** [reply]
what is the solution then

**Reply by DemonJAZ** [reply]
> what is the solution then

Delete old brew installation and install homebrew fresh

**Reply by riturajratan** [reply]
still same issue

**Reply by damuz91** [reply]
Im having the same issue too. I have removed and reinstalled homebrew but no luck

**Comment by damuz91** [comment]
```
arch -x86_64 brew list
```

The arch -x86_64 part specifies the Intel architecture. If this command works without errors, Homebrew is using the Intel version.

**Reply by ortonomy** [reply]
nah, because `arch -arm64 brew list` works without errors, too....

**Reply by nikTC** [reply]
I was facing same issue, and specifiying the architecture worked for me.
`arch -x86_64 rbenv install 3.3.5`

**Comment by usingh282** [comment]
Any update here BUILD FAILED (macOS 14.5 on arm64 using ruby-build 20241017)??

**Reply by Timi-d1** [reply]
@usingh282 end up figuring it out?
